pkg/api/alerting.go

```diff
@@ -179,8 +180,13 @@ func GetAlert(c *models.ReqContext) response.Response {
 	return response.JSON(200, &query.Result)
 }
 
-func GetAlertNotifiers(c *models.ReqContext) response.Response {
-	return response.JSON(200, alerting.GetNotifiers())
+func GetAlertNotifiers(ngalertEnabled bool) func(*models.ReqContext) response.Response {
+	return func(_ *models.ReqContext) response.Response {
+		if ngalertEnabled {
+			return response.JSON(200, notifier.GetAvailableNotifiers())
+		}
+		return response.JSON(200, alerting.GetNotifiers())
```

Comment:
Shouldn't this be a 501 not implemented in the case of ngalert disabled? I don't think we'll continue to support the old alerting path in v8 so might as well remove it.

Reply:
Fair enough. Let's add a TODO or a [cookie](https://www.youtube.com/watch?v=iMSBuNaJh7A) to remind ourselves as we clean up then.